galsim/catalog.py

```diff
@@ -343,3 +349,255 @@ def __hash__(self): return hash(repr(self))
+    def __init__(self, names, types=None, _rows=None, _sort_keys=None):
+        self.names = names
+        if types is None:
+            self.types = [ float for i in names ]
+        else:
+            self.types = types
+        if _rows is None:
+            self.rows = []
+        else:
+            self.rows = _rows
+        if _sort_keys is None:
+            self.sort_keys = []
+        else:
+            self.sort_keys = _sort_keys
+
+    @property
+    def nobjects(self): return len(self.rows)
+    @property
+    def ncols(self): return len(self.names)
+
+    # Again, when we use this through a proxy, we need getters for the attributes.
+    def getNames(self): return self.names
+    def getTypes(self): return self.types
+    def setTypes(self, types): self.types = types
+    def getNObjects(self): return self.nobjects
+    def getNCols(self): return self.ncols
+
+    def add_row(self, row, sort_key=None):
+        """Add a row of data to the catalog.
+
+        Warning: no type checking is done at this point.  If the values in the row do not
+        match the column types, you may get an error when writing, or you may lose precision,
+        depending on the nature of the mismatch.
+
+        @param row          A list with one item per column in the same order as the names list.
+        @param sort_key     If the rows may be added out of order, you can provide a sort_key,
+                            which will be used at the end to re-sort the rows.
+        """
+        if len(row) != self.ncols:
+            raise ValueError("Length of row does not match the number of columns")
+        self.rows.append(tuple(row))
+        if sort_key is None:
+            self.sort_keys.append(self.nobjects)
+        else:
+            self.sort_keys.append(sort_key)
+
+    def write(self, file_name, dir=None, file_type=None, prec=8):
+        """Write the catalog to a file.
+
+        @param file_name    The name of the file to write to.
+        @param dir          Optionally a directory name can be provided if `file_name` does not 
+                            already include it. [default: None]
+        @param file_type    Which kind of file to write to. [default: determine from the file_name
+                            extension]
+        @param prec         Output precision for ASCII. [default: 8]
+        """
+        if dir is not None:
+            import os
+            file_name = os.path.join(dir,file_name)
+
+        # Figure out which file type the catalog is
+        if file_type is None:
+            import os
+            name, ext = os.path.splitext(file_name)
+            if ext.lower().startswith('.fit'):
+                file_type = 'FITS'
+            else:
+                file_type = 'ASCII'
+        file_type = file_type.upper()
+        if file_type not in ['FITS', 'ASCII']:
+            raise ValueError("file_type must be either FITS or ASCII if specified.")
+        self.file_type = file_type
```

Comment:
Why set an instance attribute here?  I don't see it used downstream, and the way you've just persisted it doesn't seem like it belongs as an attribute of the in-memory object.